### DISCUSSION TITLE
Adjoint optimization: arbitrary phase profile

### DISCUSSION THREAD

**Discussion by Liamfallik** [NanoComp/meep#2582] [Q&A]
Hi Meep community,

I am using Meep's ajoint optimizations to design metalenses and would like to design one that would produce an arbitrary phase profile that I would specify, similar to this figure from [Mansouree et al](https://www.researchgate.net/publication/348589719_Large-scale_parameterized_metasurface_design_using_adjoint_optimization):
![image](https://github.com/NanoComp/meep/assets/81376756/2ff4578c-600a-4e6a-801d-a146e65877a5)

I am currently trying to adapt the [tutorial](https://nbviewer.org/github/NanoComp/meep/blob/master/python/examples/adjoint_optimization/05-Near2Far.ipynb) to do this, with the objective function being the overlap between the resulting and the specified fields.

Is it achievable with Meep? How could I do it?
Thanks!

**Comment by stevengj** [comment]
Yes, it should be possible.
